backend/accounts/account.go

```diff
@@ -24,6 +24,11 @@ import (
 	"github.com/digitalbitbox/bitbox-wallet-app/backend/signing"
 )
 
+// Addresses is a list of addresses.
+type Addresses struct {
+	Addresses []Address
```

Comment:
How about this:

```go
type AddressList []Address
GetUnusedReceiveAddresses() []AddressList

unused := make([]accounts.AddressList, len(account.subaccounts))
for i, sub := range account.subaccounts {
    for n, addr := range sub.receiveAddresses.GetUnused() {
        if n >= receiveAddressesLimit {
            // Limit receive addresses gap to avoid longer scans or "lost" funds during a recovery.
            break
        }
        unused[i] = append(unused[i], addr)
    }
}
```

Reply:
It's just a bit hard to read Addresses[i].Addresses.Addresses.Addresses...

Reply:
I made it a struct so we can add more info to the struct if the frontend needs it, such as the name to show for the address type, or script type identifier, or sth like that.

premature?

Reply:
IMHO a bit premature, yeah.

Reply:
:+1: done